### PR TITLE
OCaml 5: resolve conflicts in stdlib mls

### PR DESCRIPTION
**Pull request by ncik-roberts**
Notes on the implementation:
  * The new function `filter_mapi` in `list.ml` is changed along the lines of our internal implementation to pass `f` as an argument rather than closing over it.
  * `stdlib/camlinternalOO.ml`: I inline the aliases to `Obj.*` functions, just like `tip-5`

Testing: I did not test, I think we'll have to wait until everything builds. David Allsopp's patches will be replayed on top of this to fix the incompatibilities with the OCaml 5 runtime, e.g. the unimplemented intrinsic `%atomic_cas`.

### CODE REVIEW

**Comment by ncik-roberts** [comment]
This just isn’t ready to review period.
It’s missing a commit. I’ll fix tomorrow.

File: stdlib/stdlib.ml [hunk 1]
```diff
@@ -83,23 +83,9 @@ external ( != ) : ('a[@local_opt]) -> ('a[@local_opt]) -> bool = "%noteq"
 
 (* Boolean operations *)
 
-<<<<<<< HEAD
-external not : (bool[@local_opt]) -> bool = "%boolnot"
-external ( & ) : (bool[@local_opt]) -> (bool[@local_opt]) -> bool = "%sequand"
 external ( && ) : (bool[@local_opt]) -> (bool[@local_opt]) -> bool = "%sequand"
 external ( or ) : (bool[@local_opt]) -> (bool[@local_opt]) -> bool = "%sequor"
```
Comment (Ekdohibs) [inline stdlib/stdlib.ml:87]: I think you removed `not` here where you wanted to remove `( or )` instead.

Comment (ncik-roberts) [reply stdlib/stdlib.ml:87]: Indeed, thanks for catching that.

File: stdlib/camlinternalOO.ml [hunk 2]
```diff
@@ -143,7 +132,7 @@ let table_count = ref 0
 
 (* dummy_met should be a pointer, so use an atom *)
 <<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (ncik-roberts) [reply stdlib/camlinternalOO.ml]: Now it is, preferring `tip-jst` as we wrap `Obj.repr` in `Sys.opaque_identity` to avoid inlining.

**Review by Ekdohibs — APPROVED** [review]

**Pull request merged**